backend/src/lib/compression.ts

```diff
@@ -7,9 +8,216 @@ import {
   createGunzip,
   createGzip,
   createInflate,
+  inflateRawSync,
+  deflateRawSync,
   Zlib,
 } from 'zlib'
 import { logger } from './logger'
+import { ServerSideEvent, WatchEvent } from './server-side-events'
+import { getEventDict } from '../routes/events'
+import { getAppDict, ICompressedResource, ITransformedResource } from '../routes/aggregators/applications'
+
+type Dictionary = {
+  arr: string[]
+  map: Record<string, string>
+  add: (key: string) => string
+  get: (inx: number) => string
+}
+
+export function createDictionary(): Dictionary {
+  const arr: string[] = []
+  const map: Record<string, string> = {}
+  const add = (key: string): string => {
+    if (!(key in map)) {
+      map[key] = `${arr.length}`
+      arr.push(key)
+    }
+    return map[key]
+  }
+  const get = (inx: number) => {
+    return arr[inx]
+  }
+  return {
+    arr,
+    map,
+    add,
+    get,
+  }
+}
+
+// keys that point to values we should filter out
+const filterKeys = new Set(['finalizers', 'flags'])
+// keys that point to unique values (don't index)
+const valueAsIsKeys = new Set(['uid', 'name', 'resourceVersion', 'generation'])
+// keys that point to values that are likely to be repeated (we should index)
+const valueInDictionaryKeys = new Set([
+  'apiVersion',
+  'kind',
+  'namespace',
+  'pathname',
+  'group',
+  'webConsoleURL',
+  'ocpClusterId',
+])
+
+type SortTimeType = {
+  lastTransitionTime?: Date
+}
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type UncompressedResourceType = Record<string, any> | Record<string, any[]> | string | number
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type CompressedResourceType = Record<number, any> | Record<number, any[]> | string | number
+
+const NUMBER_MARKER = '#!%'
+
+export function deflateResource(resource: IResource, dictionary: Dictionary): Buffer {
+  const res = compressResource(resource as UncompressedResourceType, dictionary)
+  return deflateRawSync(JSON.stringify(res))
+}
+
+function compressResource(
+  resource: UncompressedResourceType,
+  dictionary: Dictionary,
+  parentKey?: string
+): CompressedResourceType {
+  if (resource) {
+    if (Array.isArray(resource)) {
+      // filter out all but the 3 most recent conditions
+      if (parentKey === 'conditions') {
+        resource = resource
+          .sort((a: SortTimeType, b: SortTimeType) => {
+            return new Date(b?.lastTransitionTime).getTime() - new Date(a?.lastTransitionTime).getTime()
+          })
+          .slice(0, 3)
+      }
```

Comment:
I'm not confident this is a safe change to make. Status conditions are used a lot in the UI code.

Reply:
Example:
<img width="873" alt="image" src="https://github.com/user-attachments/assets/451e38f0-3e2a-4898-9e91-749f63e69de9" />